mindsdb/integrations/handlers/s3_handler/README.md

```diff
@@ -42,10 +42,24 @@ Retrieve data from a specified object (file) in the S3 bucket by providing the i
 
 ```sql
 SELECT *
-FROM s3_datasource.`my-file.csv`;
-LIMIT 10;
+FROM s3_datasource.`my-file.csv` LIMIT 10
 ```
 
+Retrieve list of files (not filtered by extension):
+
+```sql
+SELECT *
+FROM s3_datasource.files LIMIT 10
+```
+
+Getting list of files with content (content column have to be requested in targets):
+
+```sql
+SELECT path, content
+FROM s3_datasource.files LIMIT 10
+```
```

Comment:
Nit: just to improve grammar,

Retrieve a list of files:

```sql
SELECT *
FROM s3_datasource.files LIMIT 10
```

Retrieve a list of files with their content (the `content` column needs to be requested explicitly):

```sql
SELECT path, content
FROM s3_datasource.files LIMIT 10
```

Reply:
Updated